src/main/java/com/google/sps/utility/LocationsUtility.java

```diff
@@ -63,4 +64,36 @@ private static String getLocation(Pattern pattern, String prefix, String taskNot
     }
     return "";
   }
+
+  /**
+   * Generates combinations of the lists and appends them to the result parameter.
+   *
+   * @param lists A list of lists to generate combinations for.
+   * @param result A pointer to a list to contain the generated combinations.
+   * @param current A list containing the combination built so far at the recursive call.
+   */
+  private static void generateCombinationsHelper(
+      List<List<String>> lists, List<List<String>> result, List<String> current) {
+    int depth = current.size();
+    if (depth == lists.size()) {
+      result.add(new ArrayList<>(current));
+      return;
+    }
+    for (int i = 0; i < lists.get(depth).size(); i++) {
+      current.add(lists.get(depth).get(i));
+      generateCombinationsHelper(lists, result, current);
+      current.remove(depth);
+    }
+  }
+
+  /**
+   * Generates all combinations of the lists recursively.
+   *
+   * @param lists A list of lists to generate combinations for.
+   */
+  public static List<List<String>> generateCombinations(List<List<String>> lists) {
```

Comment:
Food for thought: if you had a non-recursive implementation, then you may be able to turn this into a combination supplier for use with `Stream.generate`. This would essentially produce combinations on-demand. When producing a potentially large (like use-all-your-memory large) result, it's nice to give the caller an option to only consume some of the elements.

(No need to make this change. I am just thinking aloud.)

Reply:
Ohhh okay, `Stream.generate` would have been neat. Also some sort of dynamic programming would have been cool but would be a lot more complicated to implement.

Reply:
`RESOLVED`